### ISSUE TITLE
Mouse spinning in games

### ISSUE THREAD

**Issue by jabuxas** [hyprwm/Hyprland#4015]
### Hyprland Version

Hyprland, built from branch HEAD at commit 9afdd61adeaf413afbe853dbc5fdf6853c8bde21  (props: bump ver to 0.32.3). Tag: v0.32.3  flags: (if any) debug

### Bug or Regression?

Bug

### Description

Mouse spins in games and camera is unusable, just moving a bit moves the camera around a whole lot. Gamescope fixes it, but I experience a lot of input lag using gamescope.

### How to reproduce

I think it has got to do with having a multi-monitor setup, not really sure apart from that. Sometimes I just leave the window in another workspace and then come back and the mouse is bugged. Other times, just moving my mouse to the 2nd monitor and then coming back bugs out.

### Crash reports, logs, images, videos

showing bug: https://youtu.be/KsKSGZCjw6g

**Comment by vaxerski** [comment]
does zis happen on sway?

**Comment by jbalme** [comment]
Confirming it happens for me in 0.30.0 in FFXIV.

**Comment by h0tc0d3** [comment]
@vaxerski I have this error in overwatch 2, 1 monitor. As I understand, the event occurs only when some application blocks the input device (for example, mouse control software), and the mouse temporarily stops transmitting position data, i.e. the mouse is busy processing codes from the control software, and does not send data about the mouse displacement. For example, for [Razer mice](https://github.com/h0tc0d3/razerctl/blob/master/src/hidraw.c#L221) the device is blocked for 30ms, because any request, for example changing DPI, requires processing of the request by the mouse processor, and this does not happen immediately. Probably, if the input device does not respond for a short time, then CInputManager::simulateMouseMovement is probably triggered, and then the InputManager stops processing some of the events from the mouse. I also noticed that my cursor is very nervous and moves on the spot, I solved this by removing the CInputManager::simulateMouseMovement call and replacing vec() with goalv() in https://github.com/hyprwm/Hyprland/blob/main/ src/managers/input/InputManager.cpp

**Comment by jabuxas** [comment]
happens on monster hunter world as well. wiggling the window works if the game is running in windowed mode, but if it runs in windowed mode it doesn't automatically get 1920x1080 resolution even when fullscreening the window.

**Comment by MightyPlaza** [comment]
> happens on monster hunter world as well. wiggling the window works if the game is running in windowed mode, but if it runs in windowed mode it doesn't automatically get 1920x1080 resolution even when fullscreening the window.

does `fakefullscreen`/`nofullscreenrequest`/`nomaximizerequest` make any differnece?

**Comment by jabuxas** [comment]
it doesn't seem to make any difference with fullscreening the window originally windowed, the then fullscreened window is still with black borders on some sides. (example with warframe)
![image](https://github.com/hyprwm/Hyprland/assets/94939040/567d1fce-1d6c-44ba-8e0c-d5a7705edffe)


about the mouse spin, it doesn't seem to be happening anymore, but it's kinda random. i put all of the 3 options on the games, i'll try for a bit and if it doesn't happen anymore and if it's considered a fix, i'll close.

**Comment by led0nk** [comment]
I kind of got the same issue.
I use a multi-monitor setup and randomly sometimes after i switched from a game to another browser workspace my mouse sensitivity is like 10 times as high as before. 

On top of that i got the issue that i cannot 360 my cameraview in e.g. Ark SE and the camera sometimes stutters like it went upside down.
Maximized-window-modus, window-modus etc. doesn't make any difference.

**Comment by quadbyte** [comment]
I'm affected too.   It happens about 20% of the time after I launch Overwatch 2,  sometimes it's easy to fix (getting in the menus doing a few mouse move and clicks).   Sometimes it's super difficult to get the focus back. Have not found a proper way to fix it.

When it's happening the character is looking straight up to the ceiling and any touch of the mouse will make it spins super fast.

**Comment by MoonBurst** [comment]
Yeah, this issue, and poor grouping (which can be mostly fixed by hy3) are the main things keeping me off Hyprland. The spinning when I'm in games. I can't figure out exactly what causes it, nor a solid fix though. As far as I can guess, it has something to do with alt+tabbing out, and maybe multi-monitor.

**Comment by jabuxas** [comment]
it is multi monitor related, but it's hard to reproduce consistently, sometimes it happens on its own, sometimes i change workspaces a lot and then it happens, sometimes it doesn't happen at all.

**Comment by Enovale** [comment]
I get this problem as well with overwatch 2, ffxiv, and lethal company as well iirc.

It happens regardless of the window mode (windowed, fullscreen, borderless) but sometimes fixes if i change between them (At random) and resize the window a bunch in floating mode (still only randomly fixes it).

Also even if its fixed it will sometimes start happening again after a focus change, especially using a hotkey to switch to the workspace from a different workspace.

**Comment by tchofy** [comment]
I wonder if https://github.com/hyprwm/Hyprland/pull/4889 fixed this issue or changed the behavior. Is anyone on a -git build able to replicate this?

**Comment by quadbyte** [comment]
I'm on a -git build of hyprland that I update daily. 
Situation has slightly improved lately, it happens less frequently. 
But the bug it's still there, I got it two days ago.

Is there any log we could turn on or provide, to help pin pointing the issue ?

**Comment by quadbyte** [comment]
For what it's worth, I also have the issue with KDE Plasma 6, playing OW2.

**Comment by h0tc0d3** [comment]
> For what it's worth, I also have the issue with KDE Plasma 6, playing OW2.
Sway does not have this problem and I use native wayland in wine 9.0 to play Overwatch 2, maybe this is a xwayland bug.

**Comment by Enovale** [comment]
This bug report describes the exact issue and it apparently got fixed in kwin: https://bugs.kde.org/show_bug.cgi?id=482476
It seems to possibly have been handled by this commit: https://invent.kde.org/plasma/kwin/-/commit/630ba5fab4b02c2b496553c79d4a315da9271b53

**Comment by MROvaiz** [comment]
Having same issue with all COD Series and NMS as of now which i have tested.
Facing this for 1 year now.

**Comment by vaxerski** [comment]
please check with above

**Comment by JonathanMathBoi** [comment]
Running on the git AUR version rn, and that seems to have fixed the problem for me. Thanks.

**Comment by vaxerski** [comment]
@jabuxas can this be closed?

**Comment by MROvaiz** [comment]
It's Still spinning in NMS, I'm on latest version 

`[mro@nixbin:~]$ hyprctl systeminfo
Hyprland, built from branch  at commit 9a66514e26319e41c8661f57097a73b1c8579ccf  ().
Date: 2024-04-16
Tag: , commits: @COMMITS@`

Here link, https://www.youtube.com/watch?v=kFkrFvZnCng

**Comment by jabuxas** [comment]
seems fine by me

**Comment by MROvaiz** [comment]
Last update from me, It's working.
Problem with Lutris or wine, i checked in steam as non steam game with proton 8.
It's working smoothly, No issues for far 3 hours playing. Please close this.

Thank you.

**Comment by Nama** [comment]
I still have this issue on 0.39.1 D:

**Comment by bljordan** [comment]
This problem still exists on hyprland-0.42.0. I've tested running from Steam and Lutris with the same results.
Gamescope may fix the problem but introduces other stability issues.

The problem does not exist in wayland Gnome for me.

Should I make a new issue or can we reopen this one?

**Comment by bljordan** [comment]
I've confirmed disabling the second monitor fixes this.
`monitor=DP-2,disable`

Obviously not a great solution, but confirming it is related.

**Comment by Nama** [comment]
Didn't have any second monitor enabled while gaming, still occurred.

**Comment by lpuv** [comment]
Unfortunately still occurs for me.

**Comment by sr-tream** [comment]
Hello, have same or similar issue - resolved by disabling explicit sync for me (nvidia gpu)

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
